### DISCUSSION TITLE
Growing Memory on Demand for WebAssembly

### DISCUSSION THREAD

**Discussion by christophwille** [open-policy-agent/community#647] [OPA and Rego]
https://github.com/christophwille/dotnet-opa-wasm/pull/58

I basically did the same as the npm version - check if memory size is sufficient, grow if necessary. All of the other stuff remained exactly the same, however, in my case it then fails inside opa_eval (enlarging the memory works, writing the string works, but then opa_eval does the same as in the "too little memory case" which is strange).

Any ideas where to start looking? (note my C# is rego_v1 whereas npm isn't yet)

**Comment by anderseknert** [comment]
Hi there! May be that there’s some delay before someone can look into this :)

Just to help us understand here, is this something you had working pre-1.0 and now it doesn’t work anymore, or just that you tried it now for the first time and it doesn’t work as you expected?

**Reply by christophwille** [reply]
The automatic memory-expansion is something new on top of an existing PR (the existing PR had no unit tests, so I was reluctant to take it)
